File: test/node/randomWalk.spec.js
describe('randomWalk', function() {
  var _ = require('underscore');
  var module = require('../../src/node/randomWalk');
  var solver;
  var room;
  var exit_1_0_0 = {
    x: 1,
    y: 0,
    z: 0
  };
  var exit_0_1_0 = {
    x: 0,
    y: 1,
    z: 0
  };

  beforeEach(function() {
    room = {
      x: 0,
      y: 0,
      z: 0
    };
    solver = module.create();
  });

  it('picks only exit', function() {
    room.exits = [exit_1_0_0];
    var location = solver.next(room);
    expect(location).toBe(exit_1_0_0);
    expect(solver.nextCalls).toBe(1);
  });

  it('picks first exit after randomizing', function() {
    spyOn(_, 'shuffle').andReturn([exit_1_0_0, exit_0_1_0]);
    room.exits = [exit_0_1_0, exit_1_0_0];
    var location = solver.next(room);
    expect(location).toBe(exit_1_0_0);
    expect(_.shuffle).toHaveBeenCalledWith(room.exits);
    expect(solver.nextCalls).toBe(1);
  });
});

File: src/test/ruby/random_walk_spec.rb
require "rubygems"
require "bundler/setup"
require File.join(File.dirname(__FILE__), "../../main/ruby/random_walk")

describe Solver::RandomWalk do
  exit_1_0_0 = {
    "x" => 1,
    "y" => 0,
    "z" => 0
  };

  exit_0_1_0 = {
    "x" => 0,
    "y" => 1,
    "z" => 0
  };

  before do
    @solver = Solver::RandomWalk.new()
    @room = {
      "x" => 0,
      "y" => 0,
      "z" => 0,
      "exits" => []
    }
  end

  it "picks only exit" do
    @room["exits"] << exit_0_1_0
    next_location = @solver.next(@room)
    expect(next_location).to equal(exit_0_1_0)
  end

  it "returns randomly chosen exit" do
    @room["exits"] << exit_0_1_0
    @room["exits"] << exit_1_0_0
    @solver.stub(:rand) { 1 }
    next_location = @solver.next(@room)
    expect(next_location).to equal(exit_1_0_0)
  end
end
